src/graph/serializers/encoder/with_server_events/v7/encode_graphcontext.c

```diff
@@ -0,0 +1,101 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "encode_v7.h"
+#include "../../../../../util/arr.h"
+
+extern bool process_is_child; // Global variable declared in module.c
+
+// Determine whether we are in the context of a bgsave, in which case
+// the process is independent and should not acquire locks.
+static inline bool _shouldAcquireLocks(void) {
+	return !process_is_child;
+}
+
+static void _RdbSaveHeader(RedisModuleIO *rdb, GraphContext *gc) {
+	/* Header format:
+	* Graph name
+	* Number of graph keys (graph context key + meta keys)
+	* Number of processed keys (current payload index)
+	* Current payload type
+	*/
+
+	// Graph name.
+	RedisModule_SaveStringBuffer(rdb, gc->graph_name, strlen(gc->graph_name) + 1);
+
+	// Number of keys
+	RedisModule_SaveUnsigned(rdb, GraphEncodeContext_GetKeyCount(gc->encoding_context));
+
+	// Payload type
+	RedisModule_SaveUnsigned(rdb, GraphEncodeContext_GetEncodePhase(gc->encoding_context));
+}
+
+// Select the first phase to encode.
+static void _SelectFirstPhase(GraphContext *gc) {
+	// If there are nodes
+	if(Graph_NodeCount(gc->g) > 0) {
+		GraphEncodeContext_SetEncodePhase(gc->encoding_context, NODES);
+	}
+	// If all nodes are deleted
+	else if(array_len(gc->g->nodes->deletedIdx) > 0) {
```

Comment:
I don't personally care for separating conditions this way, and would instead do
```
} else if(array_len(gc->g->nodes->deletedIdx) > 0) {
	// If all nodes are deleted
```
Or return within each condition and have each condition be an `if`.
This is an aesthetic preference, though, not a mandate!